ESPnet-SPK: add ASVspoof19 SASV recipe

Alexgichamba opened the pull request:
## What?

Add a spk1 recipe on the ASVspoof19 Logical Access ASV subset.

## Why?

Provide diverse recipes for ESPnet-SPK, adding an SASV example.

## See also

[ASVspoof19](https://www.asvspoof.org/index2019.html)

sw005320 commented:
@Jungjee, can you review this?

Jungjee commented on egs2/TEMPLATE/spk1/db.sh:
```diff
@@ -1,6 +1,7 @@
 # Set the path of your corpus
 # "downloads" means the corpus can be downloaded by the recipe automatically
 
+ASVSpoof_LA=downloads
```
Could you change the name to ASVSPOOF2019_LA or ASVSpoof2019_LA?

Jungjee commented on egs2/asvspoof/spk1/local/asv_data_prep.py:
```diff
@@ -0,0 +1,48 @@
+# asv_data_prep.py
```
could you replace this line with a brief explanation what this script does?

Jungjee commented on egs2/asvspoof/spk1/local/asv_data_prep.py:
```diff
@@ -0,0 +1,48 @@
+    ) as f_utt2spk, open(os.path.join(dst, "wav.scp"), "w") as f_wav:
+
+        for spk in spk2utt:
+            f_spk2utt.write(f"{spk} {' '.join(spk2utt[spk])}\n")
```
note that there's an utt2spk to spk2utt script within ESPnet.

Jungjee commented on egs2/asvspoof/spk1/local/cm_data_prep.py:
```diff
@@ -0,0 +1,39 @@
+#!/usr/bin/env bash
```
[question] Where do we need this? 
Do we need the train and the validation sets as well?

Alexgichamba replied:
No. I did not use this. I thought perhaps someone using the recipe might want to do some in-domain finetuning. I could take it out if it is not needed.

Jungjee commented on egs2/asvspoof/spk1/local/data.sh:
```diff
@@ -0,0 +1,222 @@
+    log "Stage 2, DONE."
+fi
+
+log "Successfully finished. [elapsed=${SECONDS}s]"
```
remove?

Jungjee commented on egs2/asvspoof/spk1/run.sh:
```diff
@@ -0,0 +1,25 @@
+spk_config=conf/train_rawnet3.yaml
+
+train_set="train"
+valid_set="test"
+cohort_set="test"
```
Please note that the cohort set cannot be equal to the test set. 
Could you use the development set since ASVspoof2019 LA explicitly has this partition?

Alexgichamba replied:
I see. Thanks for making me aware of this. I have added support for the dev partition, and I've assigned it to be the valid set and cohort set

Jungjee reviewed (commented):
Thank you for the work Alex!
I left a few comments, but in general it looks great!

Alexgichamba commented:
> Thank you for the work, Alex! I left a few comments, but in general, it looks great!
> 
> Meanwhile, please make additional required updates to make all CI tests pass.

Hi @Jungjee ,
Thanks for the review. I will address the comments.

Alexgichamba commented:
Hi @Jungjee,
I have addressed the comments, except for two of them.

1. Should I remove the CM data prep? I had figured it might be useful later for in-domain finetuning
2. For the utt2spk/spk2utt, should I change it and use the script instead?
3. The CI checks that I am failing seem to be from install_espnet or test_python/test/espnet2/gan_tts. I am not quite sure how to make these pass

sw005320 commented:
> 3. The CI checks that I am failing seem to be from install_espnet or test_python/test/espnet2/gan_tts. I am not quite sure how to make these pass

I reran the CI

Alexgichamba commented:
> > 3. The CI checks that I am failing seem to be from install_espnet or test_python/test/espnet2/gan_tts. I am not quite sure how to make these pass
> 
> I reran the CI

Thank you. Seems to be okay now

sw005320 commented:
> Thanks @Alexgichamba for the updates and your effort!!
> 
> > 1. Should I remove the CM data prep? I had figured it might be useful later for in-domain fine-tuning
> 
> I see. For me, it's okay to leave, but not so sure how @sw005320 would think about this.
> 

We can leave it, then.

sw005320 commented:
Thanks, @Alexgichamba
If you have some follow-up activities, please make a PR.

Alexgichamba commented:
> Thanks, @Alexgichamba If you have some follow-up activities, please make a PR.

Thanks @sw005320 and @Jungjee. I appreciate the support!
